src/workerd/api/basics.c++

```diff
@@ -521,13 +522,96 @@ bool EventTarget::dispatchEvent(jsg::Lock& js, jsg::Ref<Event> event) {
   return dispatchEventImpl(js, kj::mv(event));
 }
 
+// A wrapper for the AbortTrigger jsrpc client, that automatically sends a release() message once
+// the client is destroyed, informing the server that an abort will not be triggered in the future.
+class AbortTriggerRpcClient final {
+ public:
+  AbortTriggerRpcClient(rpc::AbortTrigger::Client&& client): client(kj::mv(client)) {}
+
+  kj::Promise<void> abort(kj::ArrayPtr<kj::byte> reason) {
+    auto req = client.abortRequest(capnp::MessageSize{reason.size() / sizeof(capnp::word) + 8, 0});
+    auto field = req.initReason();
+    field.setV8Serialized(reason);
+    return req.send().ignoreResult();
+  }
+
+  ~AbortTriggerRpcClient() noexcept(false) {
+    if (skipReleaseForTest) {
+      return;
+    }
+
+    auto req = client.releaseRequest(capnp::MessageSize{4, 0});
+    // We call detach() on the resulting promise so that we can perform RPC in a destructor
+    req.send().ignoreResult().detach([](kj::Exception exc) {
+      if (exc.getType() == kj::Exception::Type::DISCONNECTED) {
+        // It's possible we can't send the release message because we're already disconnected.
+        return;
+      };
+
+      // Other exceptions could be more interesting
+      LOG_EXCEPTION("~AbortTriggerRpcClient", exc);
+    });
+  }
+
+  bool skipReleaseForTest = false;
+
+ private:
+  rpc::AbortTrigger::Client client;
+};
+
+namespace {
+// The jsrpc handler that receives aborts from the remote and triggers them locally
+class AbortTriggerRpcServer final: public rpc::AbortTrigger::Server {
+ public:
+  AbortTriggerRpcServer(kj::Own<kj::PromiseFulfiller<void>> fulfiller,
+      kj::Own<AbortSignal::PendingReason>&& pendingReason)
+      : fulfiller(kj::mv(fulfiller)),
+        pendingReason(kj::mv(pendingReason)) {}
+
+  kj::Promise<void> abort(AbortContext abortCtx) override {
+    auto params = abortCtx.getParams();
+    auto reason = params.getReason().getV8Serialized();
+
+    pendingReason->getWrapped() = kj::heapArray(reason.asBytes());
+    fulfiller->fulfill();
+    return kj::READY_NOW;
+  }
+
+  kj::Promise<void> release(ReleaseContext releaseCtx) override {
+    released = true;
+    return kj::READY_NOW;
+  }
+
+  ~AbortTriggerRpcServer() noexcept(false) {
+    if (pendingReason->getWrapped() != nullptr) {
+      // Already triggered
+      return;
+    }
+
+    if (!released) {
+      pendingReason->getWrapped() = kj::Exception(kj::Exception::Type::FAILED, __FILE__, __LINE__,
+          kj::str(JSG_ERROR_DOMAbortError ": Dropped capability, assuming abort"));
```

Comment:
This message probably isn't understandable to most Workers developers. Perhaps:

> An AbortSignal received over RPC was implicitly aborted because the connection back to its trigger was lost.